Framework/PythonInterface/plugins/functions/TeixeiraWater.py

```diff
@@ -31,9 +33,21 @@ def init(self):
     def function1D(self, xvals):
         tau = self.getParameterValue("Tau")
         length = self.getParameterValue("L")
-        q2l2 = np.square(length * np.array(xvals))
-        return (self.hbar/tau) * q2l2 / (6 + q2l2)
+        xvals = np.array(xvals)
+
+        with np.errstate(divide='ignore'):
+            hwhm = self.hbar * xvals * xvals * length / (tau * (1 + xvals * xvals * length))
```

Comment:
Is this correct? In the documentation, `length` is also squared. And by the way, using `numpy.square()` might be considered here (and other parts of this PR as well).

Reply:
Also, the documentation says 6 + (QL)^2 instead of 1 + Q^2 * L

Reply:
@SpencerHowells is the documentation correct here?

Reply:
@soininen 6 + (QL)^2 is correct, thanks for spotting this!